pinot-core/src/test/java/org/apache/pinot/core/accounting/PerQueryCPUMemAccountantFactoryTest.java

```diff
@@ -0,0 +1,269 @@
+        Boolean.toString(EXPECTED_OOM_KILL_QUERY_ENABLED));
+    CLUSTER_CONFIGS.put(getFullyQualifiedConfigName(CommonConstants.Accounting.CONFIG_OF_PUBLISHING_JVM_USAGE),
+        Boolean.toString(EXPECTED_PUBLISH_HEAP_USAGE_METRIC));
+    CLUSTER_CONFIGS.put(
+        getFullyQualifiedConfigName(CommonConstants.Accounting.CONFIG_OF_CPU_TIME_BASED_KILLING_ENABLED),
+        Boolean.toString(EXPECTED_IS_CPU_TIME_BASED_KILLING_ENABLED));
+    CLUSTER_CONFIGS.put(
+        getFullyQualifiedConfigName(CommonConstants.Accounting.CONFIG_OF_CPU_TIME_BASED_KILLING_THRESHOLD_MS),
+        Long.toString(EXPECTED_CPU_TIME_BASED_KILLING_THRESHOLD_NS));
+    CLUSTER_CONFIGS.put(getFullyQualifiedConfigName(CommonConstants.Accounting.CONFIG_OF_PANIC_LEVEL_HEAP_USAGE_RATIO),
+        Double.toString(EXPECTED_PANIC_LEVEL));
+    CLUSTER_CONFIGS.put(
+        getFullyQualifiedConfigName(CommonConstants.Accounting.CONFIG_OF_CRITICAL_LEVEL_HEAP_USAGE_RATIO),
+        Double.toString(EXPECTED_CRITICAL_LEVEL));
+    CLUSTER_CONFIGS.put(getFullyQualifiedConfigName(
+            CommonConstants.Accounting.CONFIG_OF_CRITICAL_LEVEL_HEAP_USAGE_RATIO_DELTA_AFTER_GC),
+        Double.toString(EXPECTED_CRITICAL_LEVEL_AFTER_GC));
+    CLUSTER_CONFIGS.put(
+        getFullyQualifiedConfigName(CommonConstants.Accounting.CONFIG_OF_ALARMING_LEVEL_HEAP_USAGE_RATIO),
+        Double.toString(EXPECTED_ALARMING_LEVEL));
+    CLUSTER_CONFIGS.put(getFullyQualifiedConfigName(CommonConstants.Accounting.CONFIG_OF_SLEEP_TIME_MS),
+        Integer.toString(EXPECTED_NORMAL_SLEEP_TIME));
+    CLUSTER_CONFIGS.put(getFullyQualifiedConfigName(CommonConstants.Accounting.CONFIG_OF_SLEEP_TIME_DENOMINATOR),
+        Integer.toString(EXPECTED_ALARMING_SLEEP_TIME_DENOMINATOR));
+    CLUSTER_CONFIGS.put(
+        getFullyQualifiedConfigName(CommonConstants.Accounting.CONFIG_OF_MIN_MEMORY_FOOTPRINT_TO_KILL_RATIO),
+        Double.toString(EXPECTED_MIN_MEMORY_FOOTPRINT_FOR_KILL));
+    CLUSTER_CONFIGS.put(getFullyQualifiedConfigName(CommonConstants.Accounting.CONFIG_OF_GC_BACKOFF_COUNT),
+        Integer.toString(EXPECTED_GC_BACKOFF_COUNT));
+    CLUSTER_CONFIGS.put(getFullyQualifiedConfigName(CommonConstants.Accounting.CONFIG_OF_GC_WAIT_TIME_MS),
+        Integer.toString(EXPECTED_GC_WAIT_TIME));
+    CLUSTER_CONFIGS.put(getFullyQualifiedConfigName(CommonConstants.Accounting.CONFIG_OF_QUERY_KILLED_METRIC_ENABLED),
+        Boolean.toString(EXPECTED_IS_QUERY_KILLED_METRIC_ENABLED));
+  }
+
+  @Test
+  void testOOMProtectionKillingQueryConfigChange() {
+    PerQueryCPUMemAccountantFactory.PerQueryCPUMemResourceUsageAccountant accountant =
+        new PerQueryCPUMemAccountantFactory.PerQueryCPUMemResourceUsageAccountant(new PinotConfiguration(), "test",
+            InstanceType.SERVER);
+
+    accountant.getWatcherTask().onChange(
+        Set.of(getFullyQualifiedConfigName(CommonConstants.Accounting.CONFIG_OF_OOM_PROTECTION_KILLING_QUERY)),
```

Comment:
nit: I'd recommend adding an assert on what the existing value should be prior to the `onChange` call to ensure that the update actually went through. It'll be good to add this assert for all tests

e.g. for this:

```
    assertFalse(accountant.getWatcherTask().getQueryMonitorConfig().isOomKillQueryEnabled());
    accountant.getWatcherTask().onChange(
        Set.of(getFullyQualifiedConfigName(CommonConstants.Accounting.CONFIG_OF_OOM_PROTECTION_KILLING_QUERY)),
        CLUSTER_CONFIGS);
    assertTrue(accountant.getWatcherTask().getQueryMonitorConfig().isOomKillQueryEnabled());
```